Implement dictionary based function signatures and a minimal app

Eisbrenner opened the pull request:
Refactor api, add app, add conversions to api

- Ignore unused import error in \_\_init\_\_.py
- Refactor api.diagnostics to get functions reusable in api.converions
- Reactivate turned-off tests
- Add .coveragerc and subsequently reduce pytest call length in tox.ini
- Merge .isort.cfg into pyproject.toml
- Update README.md and add meta information to pyproject.toml
- Update pyproject.toml and poetry.lock file

Eisbrenner committed:
Implement dict based func-signature in io.products

* Implement new dictionary based function signature in
windeval.io.products which contains open_products and save_products
* Update docs
* Fix minor issue with mypy in pre-commit hooks
* Add NetCDF to package dependencies
* Update poetry.lock file

Eisbrenner committed:
Implement function name convention and signatures

* Add pdbpp and ipdb to dev dependencies
* Update poetry.lock
* Fix mypy.ini path issue and add related __init__.py files in tests
* Rename exposed functions according pseudo code example
* Add API/wrapper for diagnostics
* Diagnostics follow dictionary oriented function signatures

Eisbrenner committed:
Fix config file issue of isort

Eisbrenner committed:
Replace modules with user interface section in doc

Eisbrenner committed:
Refactor api, add app, add conversions to api

- Ignore unused import error in __init__.py
- Refactor api.diagnostics to get functions reusable in api.converions
- Reactivate turned-off tests
- Add .coveragerc and subsequently reduce pytest call length in tox.ini
- Merge .isort.cfg into pyproject.toml
- Update README.md

willirath commented on pyproject.toml:
```diff
@@ -3,13 +3,19 @@ name = "windeval"
 version = "0.1.0"
```
Should we switch to calver here? I don't think, we'll need semver, because we don't  really make promises about a stable API.

Eisbrenner replied:
Yes, lets do that.

Eisbrenner replied:
I did implement a CalVer version now, I decided to simply use `YYYY.MINOR` (i.e., `2020.1` for now).

willirath commented on src/windeval/app.py:
```diff
@@ -0,0 +1,72 @@
+        paths[i] = (
+            Path(__file__)
+            .resolve()
+            .parent.parent.parent.joinpath(
```
Installing with `pip` in an otherwise empty conda env, this results in the app looking for data somewhere in the python standard lib.

Eisbrenner replied:
That was to be expected, unfortunate that the data is not copied there, I try to find an alternative, in the meantime, I'll allow the exists-check to fail.

willirath replied:
See the catalog repo for an example with included additional files: https://github.com/windeval/windeval-catalog/blob/master/pyproject.toml

Eisbrenner replied:
using the app in a "normal" environment should work now, it should just wait for you to input a valid data path before running the rest of the app.

We can adapt the behavior if desired.

willirath reviewed (commented):
Looks good. I'll see how we can tackle (test) data handling with Intake. This should also get rid of the path-resolution issues for the app.

willirath commented:
Also, see #2 (and feel free to merge there) for fixing CI here as well.

Eisbrenner committed:
Merge remote-tracking branch 'upstream/master'

Eisbrenner committed:
Fix app() test data loading path error

Running app() in regular environment won't throw an error anymore.
Instead, the app waits for input before pursuing.

Eisbrenner committed:
Introducing CalVer versioning and update pyproject

Update version number from SemVer (0.1.0) to CalVer YYYY.MINOR (2020.1)
Update pyproject.toml dependencies and introduce optional dependencies (poetry-extras)

Eisbrenner committed:
Add windeval-catalog to pyproject dependencies

Eisbrenner committed:
Pin poetry version to 1.0.5 in test workflow

Eisbrenner committed:
Add editorconfig file and update poetry behavior

+ Add pyenv local 3.7.7
+ Add .editorconfig see https://editorconfig.org/
+ Add poetry.toml to have virtual environments placed in the repo

Eisbrenner committed:
Bump poetry version in github-workflows

Eisbrenner committed:
Update function signatures and add minimal app (#2)

+ Implement dict based func-signature in `io.products`
  * Implement new dictionary based function signature in
windeval.io.products which contains `open_products` and `save_products`
  * Update docs
  * Fix minor issue with mypy in pre-commit hooks
  * Add NetCDF to package dependencies
  * Update `poetry.lock` file
+ Implement function name convention and signatures
  * The `diagnostics` module now follows dictionary oriented function signatures
  * Add pdbpp and ipdb to dev dependencies
  * Update `poetry.lock`
  * Fix `mypy.ini` path issue and add related `__init__.py` files in tests
  * Rename exposed functions according pseudo code example
  * Add API/wrapper for diagnostics
+ The windeval app can now be called from the command line with `windeval`
+ The windeval app now uses https://github.com/windeval/windeval-catalog
+ Refactor api and add conversions to api 
  * Ignore unused import error in `__init__.py`
  * Refactor `api.diagnostics` to get functions reusable in `api.conversions`
  * Reactivate turned-off tests
+ Convert poetry extras into regular dev deps
  + Poetries extras seem to be broken python-poetry/poetry#2073
